File: 0020-valid-parentheses/0020-valid-parentheses.cpp
class Solution {
public:
    bool isValid(string s) {
        stack<char> st;
        unordered_map<char,char> mpp = {{')','('},{']','['},{'}','{'}};
        for(char ch:s){
            if(ch == '(' || ch== '{' || ch=='['){
                st.push(ch);
            }else{
                char opposite = mpp[ch];
                if(st.empty()) return false;
                 if(st.top() == opposite){
                    st.pop();
                }else{
                    return false;
                }
            }
        }
        return st.empty();
    }
};

File: 0020-valid-parentheses/0020-valid-parentheses.java
class Solution {
    public boolean isValid(String s) {
        Stack<Character> st = new Stack<>();
       Map<Character, Character> mpp = Map.of(
        ')', '(',
        ']', '[',
        '}', '{'
    );
        for(Character it : s.toCharArray()){
            if(it == '{' || it == '(' || it == '['){
                st.push(it);
            }else{
                if(st.isEmpty()) return false;
                if(st.peek() != mpp.get(it)) return false;
                st.pop() ;
            }
        }

        return st.isEmpty();

    }
}